File: app/src/main/java/ru/dev2dev/sw_android/PersonActivity.java
package ru.dev2dev.sw_android;

import android.content.res.Configuration;
import android.os.Bundle;
import android.support.annotation.Nullable;

public class PersonActivity extends BaseActivity {
    public static final String EXTRA_PERSON = "ru.dev2dev.sw_android.PERSON";

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        if (getResources().getConfiguration().orientation == Configuration.ORIENTATION_LANDSCAPE) {
            finish();
            return;
        }

        PersonFragment personFragment =
                (PersonFragment) getSupportFragmentManager().findFragmentById(R.id.person_container);
        if (personFragment == null) {
            Person person = (Person) getIntent().getSerializableExtra(EXTRA_PERSON);
            personFragment = PersonFragment.newInstance(person);
            getSupportFragmentManager().beginTransaction()
                    .add(android.R.id.content, personFragment)
                    .commit();
        }
    }
}

File: app/src/main/kotlin/ru/dev2dev/sw_android/PersonActivity.kt
package ru.dev2dev.sw_android

import android.graphics.Color
import android.os.Bundle
import android.util.Log
import kotlinx.android.synthetic.main.activity_person.*

class PersonActivity : BaseActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_person)

        val person = intent.getSerializableExtra(EXTRA_PERSON) as Person
        val (name, height, mass, eyeColor, birthYear, gender) = person

        tvName.text = name
        tvGender.text = gender
        tvBirth.text = birthYear

        val info = getString(R.string.info_format, height, mass, eyeColor)
        tvInfo.text = info

        try {
            val color = Color.parseColor(eyeColor)
            ivPortrait.setBackgroundColor(color)
        } catch (e: IllegalArgumentException) {
            Log.e(TAG, "onCreate: ", e)
        }
    }

    companion object {
        val EXTRA_PERSON = "ru.dev2dev.sw_android.PERSON"
    }
}
